core/src/test/scala/dagr/core/tasksystem/PipeTest.scala

```diff
@@ -54,22 +54,21 @@ class PipeTest extends UnitSpec {
   case class BgUnzip()  extends Pipe[Binary,Text] with TestResources { def args = "bgzip" :: "--decompress" :: "--stdout" :: Nil  }
 
   "Pipe" should "connect some simple tasks" in {
-    val `|` = " " + Pipes.PipeString + " "
+    val `|` = Pipes.PipeString
     val pipe = Cat("foo.txt") | MakeCsv()
-    pipe.args.length shouldBe 5
-    pipe.args.mkString(" ") shouldBe "cat foo.txt " + Pipes.PipeString + " sed s/ +/,/g"
+    pipe.commandLine shouldBe "cat foo.txt" + Pipes.PipeString + "sed 's/ +/,/g'"
 
     val pipe2 = Cat("bar.txt") | MakeCsv() | CsvToTsv() | Column()
-    pipe2.args.mkString(" ") shouldBe "cat bar.txt" + `|` + "sed s/ +/,/g" + `|` + """tr ',' '\t'""" + `|` + "column -t"
+    pipe2.commandLine shouldBe "cat bar.txt" + `|` + "sed 's/ +/,/g'" + `|` + """tr , '\t'""" + `|` + "column -t"
   }
 
   it should "handle re-use of parts of pipes" in {
-    val `|` = " " + Pipes.PipeString + " "
+    val `|` = Pipes.PipeString
     val start = Cat("foo.txt") | MakeCsv()
     val pipe1 = start | CsvToTsv()
     val pipe2 = start | Column()
-    pipe1.args.mkString(" ") shouldBe "cat foo.txt" + `|` + "sed s/ +/,/g" + `|` + """tr ',' '\t'"""
-    pipe2.args.mkString(" ") shouldBe "cat foo.txt" + `|` + "sed s/ +/,/g" + `|` + "column -t"
+    pipe1.commandLine shouldBe "cat foo.txt" + `|` + "sed 's/ +/,/g'" + `|` + """tr , '\t'"""
```

Comment:
Change `"sed 's/ +/,/g'"` to `"sed", "'s/ +/,/g'"` here and elsewhere since I don't want to quote an arg if it starts and ends with `'`.